diff --git a/android/CMakeLists.txt b/android/CMakeLists.txt
index 1234567..abcdefg 100644
--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -41,7 +41,7 @@ if(${RNS_NEW_ARCH_ENABLED})
     if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
         target_link_libraries(rnscreens
             ReactAndroid::reactnative
             ReactAndroid::jsi
             fbjni::fbjni
             android
+            c++_shared
         )
     else()
         target_link_libraries(
             rnscreens
                 ReactAndroid::jsi
                 ReactAndroid::react_nativemodule_core
                 ReactAndroid::react_utils
                 ReactAndroid::reactnativejni
                 ReactAndroid::fabricjni
                 ReactAndroid::react_debug
                 ReactAndroid::react_render_core
                 ReactAndroid::runtimeexecutor
                 ReactAndroid::react_render_graphics
                 ReactAndroid::rrc_view
                 ReactAndroid::yoga
                 ReactAndroid::rrc_text
                 ReactAndroid::glog
                 ReactAndroid::react_render_componentregistry
                 ReactAndroid::react_render_consistency
                 ReactAndroid::react_performance_timeline
                 ReactAndroid::react_render_observers_events
                 fbjni::fbjni
                 android
+                c++_shared
         )
     endif()
 else()
     target_link_libraries(rnscreens
         ReactAndroid::jsi
         android
+        c++_shared
     )
 endif()
